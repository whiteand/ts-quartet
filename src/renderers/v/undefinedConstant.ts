import { ISchemaRenderer } from "../../types";
import { fromValueSchemaAlloc } from "./utils";

export const undefinedConstantRenderer: ISchemaRenderer<
  undefined
> = fromValueSchemaAlloc(
  valueId => `${valueId} === undefined`,
  valueId => `${valueId} !== undefined`
);
