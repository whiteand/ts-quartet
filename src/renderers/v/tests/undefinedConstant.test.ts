import { undefinedConstantRenderer } from "../undefinedConstant";
import { snapshot } from "./snapshot";

describe("v constant number renderer", () => {
  test("v(undefined)", () => {
    snapshot(undefinedConstantRenderer, { schema: undefined });
  });
});
