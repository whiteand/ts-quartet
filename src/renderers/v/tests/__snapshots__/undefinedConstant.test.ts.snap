// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v constant number renderer v(undefined) 1`] = `"_valueId === undefined"`;

exports[`v constant number renderer v(undefined) 2`] = `"_valueId !== undefined"`;

exports[`v constant number renderer v(undefined) 3`] = `"if (_valueId === undefined) return true;"`;

exports[`v constant number renderer v(undefined) 4`] = `"if (_valueId !== undefined) return false;"`;
